### PR TITLE
Fixes to get CI to build clean

### PR DESCRIPTION
**Pull request by rgulewich**
Clean up CI linting issues:

- Bring in changes to update to Go 1.11 from the `upgrade-to-go1.11` branch
- Explicitly require the gometalinter 2.0.11 release, which fixes us to specific linter versions (before this, we would get the latest gometalinter, and whatever versions it would pull in with it)
  - Builder Dockerfiles now install the gometalinter binary releases (which includes the linters it calls out to)
  - Linting changes to support the linters shipped
  - Add a convenience script for installing this version on OSX
- Don't try to install gometalinter as part of the Makefile: just die if it's not present in PATH
- Doc and comment fixes
- Added logging to various build scripts to make it clearer what steps have been run in the build

### CODE REVIEW

File: hack/install-gometalinter.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
```
Comment (sargun) [inline hack/install-gometalinter.sh]: Any reason not to use the gometalinter provided install.sh? https://github.com/alecthomas/gometalinter/blob/master/scripts/install.sh

Comment (sargun) [reply hack/install-gometalinter.sh]: Or at least wrap it?

File: uploader/s3.go [hunk 2]
```diff
@@ -76,7 +76,7 @@ func getEC2Region() (string, error) {
 func (u *S3Uploader) Upload(ctx context.Context, local string, remote string, ctypeFunc ContentTypeInferenceFunction) error {
 	u.log.Printf("Attempting to upload file from: %s to: %s", local, path.Join(u.bucketName, remote))
 
-	f, err := os.Open(local)
+	f, err := os.Open(local) // nolint: gosec
```
Comment (sargun) [inline uploader/s3.go]: Any idea why you had to add an additional gosec here?

**Comment by sargun** [comment]
LGTM. Do you want to consider using the gometalinter provided installer? And cleaning up your commits?

File: hack/install-gometalinter.sh (hunk 1 above)
Comment (rgulewich) [reply hack/install-gometalinter.sh]: Not at all - just didn't realize that it existed.  Will remove this file and point the instructions at the gometalinter version.

**Comment by rgulewich** [comment]
@sargun Updated to use the gometalinter installer.  I was thinking of doing a squash merge, or do you mean something else by cleaning up commits?

File: .buildkite/build.sh [hunk 3]
```diff
@@ -1,6 +1,10 @@
 #!/usr/bin/env bash
 set -uex -o pipefail
 
+log() {
```
Comment (sargun) [inline .buildkite/build.sh]: This is cool.

File: Makefile [hunk 4]
```diff
@@ -17,6 +17,7 @@ LOCAL_DIRS            = $(shell govendor list -p -no-status +local)
 TEST_FLAGS            ?= -v -parallel 32
 TEST_OUTPUT           ?= test.xml
 TEST_DOCKER_OUTPUT    ?= test-standalone-docker.xml
+GOVENDOR              := $(GOPATH)/bin/govendor
```
Comment (sargun) [inline Makefile]: This is pretty neat.

Comment (sargun) [reply Makefile]: Is there a way we can ensure that govendor is at least version X? Maybe have it depend on vendor/vendor.json, and make it go get -u ...?

File: Makefile [hunk 5]
```diff
@@ -130,15 +131,11 @@ clean-proto-defs: | $(clean)
 goimports:
```
Comment (sargun) [inline Makefile]: Do you want to do the same thing you did for govendor to goimports?

File: README.md [hunk 6]
```diff
@@ -24,9 +24,8 @@ export PATH=${GOPATH}/bin:${PATH}
 And then you can check out `github.com/Netflix/titus-executor` to `${GOPATH}/src/github.com/Netflix/titus-executor` -- an easy way to do this is go run `go get -u github.com/Netflix/titus-executor`.
 
 # Initial setup steps
-1. go get -u github.com/alecthomas/gometalinter
-2. gometalinter --install
-3. go get -u github.com/kardianos/govendor
+1. `./hack/install-gometalinter.sh`
+2. go get -u github.com/kardianos/govendor
```
Comment (sargun) [inline README.md]: We install this, via the makefile automagically, now, don't we?

File: executor/runtime/docker/seccomp/seccomp.go [hunk 7]
```diff
@@ -228,9 +228,9 @@ type bintree struct {
 }
 
 var _bintree = &bintree{nil, map[string]*bintree{
-	"default.json":          &bintree{defaultJson, map[string]*bintree{}},
-	"fuse-container.json":   &bintree{fuseContainerJson, map[string]*bintree{}},
-	"nested-container.json": &bintree{nestedContainerJson, map[string]*bintree{}},
+	"default.json":          {defaultJson, map[string]*bintree{}},
```
Comment (sargun) [inline executor/runtime/docker/seccomp/seccomp.go]: As an aside, I filed https://github.com/Netflix/titus-executor/issues/180, which should make this go away in the future.

File: hack/agent/Dockerfile [hunk 8]
```diff
@@ -29,22 +29,11 @@ RUN apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 581
...
-    goRelSha256='72d820dec546752e5a8303b33b009079c15c2390ce76d67cf514991646c6127b'; \
-	url="https://golang.org/dl/go${GOLANG_VERSION}.linux-amd64.tar.gz"; \
-	wget -O go.tgz "$url"; \
-	echo "${goRelSha256} *go.tgz" | sha256sum -c -; \
-	tar -C /usr/local -xzf go.tgz; \
-	rm go.tgz; \
-	export PATH="/usr/local/go/bin:$PATH"; \
-	go version
-
+ENV GOLANG_VERSION 1.10.4
```
Comment (sargun) [inline hack/agent/Dockerfile]: This was a dumb mistake on my part, you might want to remove this.

File: hack/builder/Dockerfile [hunk 9]
```diff
@@ -7,22 +7,11 @@ RUN export DEBIAN_FRONTEND=noninteractive && apt-get update && \
...
-    goRelSha256='72d820dec546752e5a8303b33b009079c15c2390ce76d67cf514991646c6127b'; \
-	url="https://golang.org/dl/go${GOLANG_VERSION}.linux-amd64.tar.gz"; \
-	wget -O go.tgz "$url"; \
-	echo "${goRelSha256} *go.tgz" | sha256sum -c -; \
-	tar -C /usr/local -xzf go.tgz; \
-	rm go.tgz; \
-	export PATH="/usr/local/go/bin:$PATH"; \
-	go version
-
+ENV GOLANG_VERSION 1.10.4
```
Comment (sargun) [inline hack/builder/Dockerfile]: This was a dumb mistake on my part, you might want to remove this.

File: hack/builder/Dockerfile [hunk 10]
```diff
@@ -40,8 +29,16 @@ RUN curl -fsSL $LIBNL_DOWNLOAD_URL -o libnl.tar.gz \
 RUN go get -u github.com/kardianos/govendor
 RUN go get -u golang.org/x/tools/cmd/goimports
 RUN go get -u github.com/mitchellh/gox
-RUN go get -u github.com/alecthomas/gometalinter
-RUN gometalinter --install
+
+# gometalinter 2.0.11:
```
Comment (sargun) [inline hack/builder/Dockerfile]: If we tell people to run hack/install-metalinter, is there a reason we don't do that ourselves?

**Comment by sargun** [comment]
There are a couple comments I have in the PR, but none of them are show-stoppers. As far as changing around the commits / cleaning them up, I wasn't thinking you should do a full-squash, just get rid of the commits made redundant.

File: hack/builder/Dockerfile (hunk 10 above)
Comment (rgulewich) [reply hack/builder/Dockerfile]: I considered it, but I got a little nervous not checksumming *something* that we download.  The gometalinter-provided install script downloads both the tarball and the checksum file from the same place, and then does the check.  If that's secure enough, then I'm game to make the change.

Comment (rgulewich) [reply hack/builder/Dockerfile]: I removed `hack/install-metalinter.sh`, by the way - the instructions now point at the one in the gometalinter repo.

File: Makefile (hunk 4 above)
Comment (rgulewich) [reply Makefile]: Good question - the best I can think of would involve using something like https://github.com/cloudflare/semver_bash to check its version.  I'll probably leave that to another PR, unless I can think of something simpler.

File: README.md (hunk 6 above)
Comment (rgulewich) [reply README.md]: Yes - removed this line.

File: Makefile (hunk 5 above)
Comment (rgulewich) [reply Makefile]: Good idea - done.

**Comment by sargun** [comment]
btw, now things are green, and you have approval, feel free to merge when you feel comfortable.

File: hack/ci-builder/Dockerfile [hunk 11]
```diff
@@ -6,44 +6,22 @@ COPY docker-repo.gpg /tmp
...
-		echo >&2 'error: UNIMPLEMENTED'; \
-		echo >&2 'TODO install golang-any from jessie-backports for GOROOT_BOOTSTRAP (and uninstall after build)'; \
-		echo >&2; \
-		exit 1; \
-	fi; \
-	\
-	export PATH="/usr/local/go/bin:$PATH"; \
-	go version
-
+COPY --from=golang:1.11-stretch /usr/local/go /usr/local/go
```
Comment (fabiokung) [inline hack/ci-builder/Dockerfile]: 🥇 💯 💯 💯 🥇

**Pull request merged**